packages/backend/src/modules/finality/analyzers/arbitrum/getSegments.ts

```diff
@@ -0,0 +1,39 @@
+import { assert } from '@l2beat/shared-pure'
+import zlib from 'zlib'
+
+import { rlpDecode } from '../../utils/rlpDecode'
+import { byteArrFromHexStr } from '../opStack/utils'
+import { blobsToData } from './blobsToData'
+import { numberToByteArr } from './utils'
+
+export function getSegments(relevantBlobs: { blob: string }[]) {
+  const blobs = relevantBlobs.map(({ blob }) => byteArrFromHexStr(blob))
+  const payload = blobsToData(blobs)
+  const decompressed = decompressPayload(payload)
+  const rlpEncoded = concatWithLengthForRlp(decompressed)
+  const segments = rlpDecode(rlpEncoded)
+  assert(Array.isArray(segments), 'Expected segments to be an array')
+
+  return segments
+}
+
+export function decompressPayload(payload: Uint8Array): Uint8Array {
+  // todo: we should check the first byte. It's probably the compression type.
+  const data = payload.slice(1)
+  const decompressedData = zlib.brotliDecompressSync(data)
+  return Uint8Array.from(decompressedData)
+}
+
+export function concatWithLengthForRlp(decompressedBytes: Uint8Array) {
```

Comment:
![image](https://github.com/l2beat/l2beat/assets/33978857/a4f05e19-f2b9-411a-ab6f-d98895d0ee3e)

Reply:
I literally have no idea, but it works.

Reply:
[chore: more comments](https://github.com/l2beat/l2beat/pull/3288/commits/2bde723ec3861ea94166cc861e64895aac846b0c)